tests/kernel/mem_slab/mslab_stats/src/main.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <ztest.h>
+
+#define BLK_SZ     64
+#define NUM_BLOCKS 8
+
+K_MEM_SLAB_DEFINE(kmslab, BLK_SZ, NUM_BLOCKS, 4);
+
+ZTEST(lib_mem_slab_stats_test, test_mem_slab_stats_invalid_params)
+{
+	struct sys_memory_stats  stats;
+	int  status;
+
+	/*
+	 * Verify that k_mem_slab_runtime_stats_get() returns -EINVAL
+	 * when an invalid parameter is passed.
+	 */
+
+	status = k_mem_slab_runtime_stats_get(NULL, &stats);
+	zassert_equal(status, -EINVAL, "Routine returned %d instead of %d",
+		      status, -EINVAL);
+
+	status = k_mem_slab_runtime_stats_get(&kmslab, NULL);
+	zassert_equal(status, -EINVAL, "Routine returned %d instead of %d",
+		      status, -EINVAL);
+
+	/*
+	 * Verify that k_mem_slab_runtime_stats_reset_max() returns -EINVAL
+	 * when an invalid parameter is passed.
+	 */
+
+	status = k_mem_slab_runtime_stats_reset_max(NULL);
+	zassert_equal(status, -EINVAL, "Routine returned %d instead of %d",
+		      status, -EINVAL);
+}
+
+ZTEST(lib_mem_slab_stats_test, test_mem_slab_runtime_stats)
```

Comment:
More of a nit (I'm not going to block this change for this) but this is testing too many things. It should really be multiple tests.